libs/fluid-elements/provider/src/lib/provider.ts

```diff
@@ -0,0 +1,232 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  html,
+  property,
+  TemplateResult,
+  css,
+  customElement,
+  LitElement,
+  CSSResult,
+  unsafeCSS,
+} from 'lit-element';
+
+import * as designTokens from '@dynatrace/fluid-design-tokens';
+
+/** Defines the possible layout density options. */
+export type FluidLayoutDensity = 'default' | 'dense' | 'loose';
+
+/** Defines the available themes. */
+export type FluidTheme = 'abyss' | 'surface';
+
+/** Type of the design tokens module */
+export type FluidDesignTokens = typeof designTokens;
+
+const supportsAdoptedStylesheets =
+  'adoptedStyleSheets' in window.ShadowRoot.prototype;
+
+// The provider itself cannot extend FluidElement
+// since it would lead to a circular dependency.
+@customElement('fluid-provider')
+export class FluidProvider extends LitElement {
+  /**
+   * Defines the theme that should be used for child components.
+   * @attr
+   * @type {'abyss' | 'surface'}
+   * @default 'abyss'
+   */
+  @property({ type: String, reflect: true })
+  theme: FluidTheme = 'abyss';
+
+  /**
+   * Defines the layout density inside the provider.
+   * @attr
+   * @type {'default' | 'dense' | 'loose'}
+   * @default 'default'
+   */
+  @property({ type: String, reflect: true })
+  layout: FluidLayoutDensity = 'default';
```

Comment:
NIT: layout is also very generic, maybe layoutDensity or layoutMode are better and more descriptive